### PR TITLE
First part of the French translation

### PR DESCRIPTION
**Pull request by KookaS**
Hello, this is a translation in French of the book. I am a native French speaker proficient in English.
It came after seeing this [issue](https://github.com/google/comprehensive-rust/issues/323).
I'll keep on working on it on a regular basis, feel free to contribute if you are good at French translation.

### CODE REVIEW

**Comment by mgeisler** [comment]
Hi @KookaS, wow, thank you very much!

I'll find someone who can skim through the PR to give is a review. I see you've marked the PR as "in progress". How complete is the translation?

**Comment by vcaen** [comment]
Hi @KookaS  and @mgeisler, happy to take a look at the PR. 

Et merci pour le travail de traduction ! :baguette_bread:

File: po/fr.po [hunk 1]
```diff
@@ -0,0 +1,15296 @@
...
+"Language: fr\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Plural-Forms: nplurals=2; plural=(n > 1);\n"
+"X-Generator: Poedit 3.2.2\n"
+
+#: src/SUMMARY.md:3
+msgid "Welcome to Comprehensive Rust 🦀"
+msgstr "Bienvenue à Rust Compréhensible 🦀"
```
Comment (vcaen) [inline po/fr.po:17]: @mgeisler Do you mean by comprehensive, do you mean "exhaustive" ? Or do you see the title as a brand, and should not be translated?

If yes to the first question, I'd make the following suggestion
```suggestion
msgstr "Bienvenue le guide complet de Rust 🦀"
```

ou 


```suggestion
msgstr "Bienvenue dans le guide exaustif de Rust 🦀"
```

File: po/fr.po [hunk 2]
```diff
@@ -0,0 +1,15296 @@
...
+
+#: src/SUMMARY.md:4
+msgid "Running the Course"
+msgstr "Suivre le cours"
```
Comment (vcaen) [inline po/fr.po:21]:
```suggestion
msgstr "Instruire le cours"
```

File: po/fr.po [hunk 3]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Keyboard Shortcuts"
+msgstr "Raccourcis de clavier"
+
+#: src/SUMMARY.md:8
+msgid "Translations"
+msgstr "Traductions"
+
+#: src/SUMMARY.md:9
+msgid "Using Cargo"
+msgstr "Utiliser cargo"
```
Comment (vcaen) [inline po/fr.po:41]:
```suggestion
msgstr "Utiliser Cargo"
```

File: po/fr.po [hunk 4]
```diff
@@ -0,0 +1,15296 @@
...
+
+#: src/SUMMARY.md:10
+msgid "Rust Ecosystem"
+msgstr "Écosystème de Rust"
+
+#: src/SUMMARY.md:11
+msgid "Code Samples"
+msgstr "Exemples de codes"
```
Comment (vcaen) [inline po/fr.po:49]:
```suggestion
msgstr "Exemples de code"
```

File: po/fr.po [hunk 5]
```diff
@@ -0,0 +1,15296 @@
...
+
+#: src/SUMMARY.md:12
+msgid "Running Cargo Locally"
+msgstr "Exécution locale"
```
Comment (vcaen) [inline po/fr.po:53]:
```suggestion
msgstr "Exécuter Cargo localement"
```

File: po/fr.po [hunk 6]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Compile Time Guarantees"
+msgstr "Garanties de temps de compilation"
+
+#: src/SUMMARY.md:25
+msgid "Runtime Guarantees"
+msgstr "Garanties d'exécution"
+
+#: src/SUMMARY.md:26
+msgid "Modern Features"
+msgstr "Caractéristiques modernes"
```
Comment (vcaen) [inline po/fr.po:89]:
```suggestion
msgstr "Fonctionnalités modernes"
```

File: po/fr.po [hunk 7]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Basic Syntax"
+msgstr "Syntaxe de base"
+
+#: src/SUMMARY.md:28
+msgid "Scalar Types"
+msgstr "Types de scalaires"
+
+#: src/SUMMARY.md:29
+msgid "Compound Types"
+msgstr "Types de composition"
```
Comment (vcaen) [inline po/fr.po:101]:
```suggestion
msgstr "Types composés"
```

File: po/fr.po [hunk 8]
```diff
@@ -0,0 +1,15296 @@
...
+
+#: src/SUMMARY.md:30
+msgid "References"
+msgstr "Références"
+
+#: src/SUMMARY.md:31
+msgid "Dangling References"
+msgstr "Références pendantes"
```
Comment (vcaen) [inline po/fr.po:109]:
```suggestion
msgstr "Références invalides"
```

File: po/fr.po [hunk 9]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Type Inference"
+msgstr "Inférence de type"
+
+#: src/SUMMARY.md:46
+msgid "static & const"
+msgstr ""
+
+#: src/SUMMARY.md:47
+msgid "Scopes and Shadowing"
+msgstr "Portée et ombrage"
```
Comment (vcaen) [inline po/fr.po:166]:
```suggestion
msgstr "Portée et masaquage"
```

File: po/fr.po [hunk 10]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Rust Memory Management"
+msgstr "Gestion de la mémoire avec Rust"
+
+#: src/SUMMARY.md:55
+msgid "Comparison"
+msgstr "Comparaison"
+
+#: src/SUMMARY.md:56
+msgid "Ownership"
+msgstr "Possession"
```
Comment (vcaen) [inline po/fr.po:202]: Propriété

File: po/fr.po [hunk 11]
```diff
@@ -0,0 +1,15296 @@
...
+
+#: src/SUMMARY.md:57
+msgid "Move Semantics"
+msgstr "Déplacer la sémantique"
```
Comment (vcaen) [inline po/fr.po:206]:
```suggestion
msgstr "Sémantique de movement"
```

File: po/fr.po [hunk 12]
```diff
@@ -0,0 +1,15296 @@
...
+
+#: src/SUMMARY.md:58
+msgid "Moved Strings in Rust"
+msgstr "String déplacées dans Rust"
```
Comment (vcaen) [inline po/fr.po:210]: avec

File: po/fr.po [hunk 13]
```diff
@@ -0,0 +1,15296 @@
...
+
+#: src/SUMMARY.md:59
+msgid "Double Frees in Modern C++"
+msgstr "Erreur de la double libération de mémoire en C++ moderne"
```
Comment (vcaen) [inline po/fr.po:214]:
```suggestion
msgstr "Double libération de mémoire en C++ moderne"
```

This is not related to an error

File: po/fr.po [hunk 14]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Field Shorthand Syntax"
+msgstr "Syntaxe abrégée des champs"
+
+#: src/SUMMARY.md:80
+msgid "Enums"
+msgstr "Énumérations"
+
+#: src/SUMMARY.md:81
+msgid "Variant Payloads"
+msgstr "Charges variantes"
```
Comment (vcaen) [inline po/fr.po:274]: Contenu variable

File: po/fr.po [hunk 15]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Method Receiver"
+msgstr "Récepteur de méthode"
+
+#: src/SUMMARY.md:85 src/SUMMARY.md:160 src/SUMMARY.md:195
+msgid "Example"
+msgstr "Exemple"
+
+#: src/SUMMARY.md:86
+msgid "Pattern Matching"
+msgstr "Correspondance de modèle"
```
Comment (vcaen) [inline po/fr.po:290]: Since model has a similar meaning as in regexp, I'd rather use "motif" than "modèle"

File: po/fr.po [hunk 16]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Destructuring Arrays"
+msgstr "Déstructuration des listes"
+
+#: src/SUMMARY.md:90
+msgid "Match Guards"
+msgstr "Gardes de match"
+
+#: src/SUMMARY.md:92
+msgid "Health Statistics"
+msgstr "Statistiques sur la santé"
```
Comment (vcaen) [inline po/fr.po:310]:
```suggestion
msgstr "Statistiques de santé"
```

File: po/fr.po [hunk 17]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Control Flow"
+msgstr "Flux de contrôle"
+
+#: src/SUMMARY.md:98
+msgid "Blocks"
+msgstr "Blocs"
+
+#: src/SUMMARY.md:99
+msgid "if expressions"
+msgstr "if expressions"
```
[the remark this answers is not in the document]
Comment (vcaen) [reply po/fr.po:330]: Similar correction needed to other "* expressions" below

File: po/fr.po [hunk 18]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Monomorphization"
+msgstr "Monomorphisation"
+
+#: src/SUMMARY.md:134
+msgid "Traits"
+msgstr "Caractéristiques"
+
+#: src/SUMMARY.md:135
+msgid "Trait Objects"
+msgstr "Objets de caractéristique"
```
Comment (vcaen) [inline po/fr.po:446]:
```suggestion
msgstr "Objets implementant des caractéristiques"
```

File: po/fr.po [hunk 19]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "A Simple GUI Library"
+msgstr "Une bibliothèque d'interface graphique"
+
+#: src/SUMMARY.md:152
+msgid "Day 3: Afternoon"
+msgstr "Jour 3 : Après-midi"
+
+#: src/SUMMARY.md:154
+msgid "Error Handling"
+msgstr "La gestion des erreurs"
```
Comment (vcaen) [inline po/fr.po:510]:
```suggestion
msgstr "Gestion des erreurs"
```

Sounds more like a title IMO

File: po/fr.po [hunk 20]
```diff
@@ -0,0 +1,15296 @@
...
+
+#: src/SUMMARY.md:155
+msgid "Panics"
+msgstr "Paniques"
+
+#: src/SUMMARY.md:156
+msgid "Catching Stack Unwinding"
+msgstr "Attraper le déroulement de la pile"
```
Comment (vcaen) [inline po/fr.po:518]: Suspendre

File: po/fr.po [hunk 21]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Documentation Tests"
+msgstr "Test de documentation"
+
+#: src/SUMMARY.md:168
+msgid "Integration Tests"
+msgstr "Test d'intégration"
+
+#: src/SUMMARY.md:169
+msgid "Unsafe Rust"
+msgstr "Rust dangereux"
```
Comment (vcaen) [inline po/fr.po:566]: Rust risqué

File: po/fr.po [hunk 22]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Extern Functions"
+msgstr "Fonctions externes"
+
+#: src/SUMMARY.md:176
+msgid "Implementing Unsafe Traits"
+msgstr "Implémentation de traits dangereux"
+
+#: src/SUMMARY.md:178
+msgid "Safe FFI Wrapper"
+msgstr "Enveloppe FFI sécurisée"
```
Comment (vcaen) [inline po/fr.po:598]: sûre

File: po/fr.po [hunk 23]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Day 4: Morning"
+msgstr "Jour 4 : Matin"
+
+#: src/SUMMARY.md:186
+msgid "Concurrency"
+msgstr "Concurrence"
+
+#: src/SUMMARY.md:187
+msgid "Threads"
+msgstr "Chaînes d’exécution"
```
Comment (vcaen) [inline po/fr.po:610]: Thread is probably well understood in French, so I would not translate this.

File: po/fr.po [hunk 24]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Android"
+msgstr "Android"
+
+#: src/SUMMARY.md:209
+msgid "Setup"
+msgstr "Installation"
+
+#: src/SUMMARY.md:210
+msgid "Build Rules"
+msgstr "Construire des règles"
```
Comment (vcaen) [inline po/fr.po:678]:
```suggestion
msgstr "Règles de construction"
```

File: po/fr.po [hunk 25]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Client"
+msgstr "Client"
+
+#: src/SUMMARY.md:219
+msgid "Changing API"
+msgstr "Modification de l'API"
+
+#: src/SUMMARY.md:220 src/SUMMARY.md:268
+msgid "Logging"
+msgstr "Enregistrement"
```
Comment (vcaen) [inline po/fr.po:718]: Journalisation

File: po/fr.po [hunk 26]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Bare Metal Rust: Afternoon"
+msgstr "Bare Metal Rust : Après-midi"
+
+#: src/SUMMARY.md:258
+msgid "Application Processors"
+msgstr "Processeurs d'applications"
+
+#: src/SUMMARY.md:259
+msgid "Inline Assembly"
+msgstr "Assemblage en ligne"
```
Comment (vcaen) [inline po/fr.po:830]: Assembleur

File: po/fr.po (hunk 1 above)
Comment (vcaen) [reply po/fr.po:17]: In that case, I would maybe consider a subtilte because Comprehsive in French is often misundertund as being "something that can be understood"

File: po/fr.po [hunk 27]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Bare Metal Rust Morning"
+msgstr "Bare Metal Rust Matin"
+
+#: src/SUMMARY.md:297
+msgid "Bare Metal Rust Afternoon"
+msgstr "Bare Metal Rust Après-midi"
+
+#: src/welcome.md:1
+msgid "# Welcome to Comprehensive Rust 🦀"
+msgstr "# Bienvenue dans Rust Compréhensible 🦀"
```
Comment (vcaen) [inline po/fr.po:938]: \"Comprehensive Rust\" following on @mgeisler comment about the title being a brand

File: po/fr.po [hunk 28]
```diff
@@ -0,0 +1,15296 @@
...
+msgid "Small Example"
+msgstr "Petit exemple"
+
+#: src/SUMMARY.md:23
+msgid "Why Rust?"
+msgstr "Pourquoi Rust ?"
+
+#: src/SUMMARY.md:24
+msgid "Compile Time Guarantees"
+msgstr "Garanties de temps de compilation"
```
Comment (sboukortt) [inline po/fr.po:81]: This is not about guarantees on build times, it’s about guarantees that are provided already when compiling. How about « Garanties à la compilation »?

File: po/fr.po (hunk 1 above)
Comment (mgeisler) [reply po/fr.po:17]: > In that case, I would maybe consider a subtilte because Comprehsive in French is often misundertund as being "something that can be understood"

Yeah, perhaps one could add a translated title in parenthesis afterwards? As translators, you're pretty free to add stuff to make things understandable :smile:

**Comment by mgeisler** [comment]
Hi all, thanks for the translation and thanks to @vcaen and @sboukortt for jumping in and reviewing this!

Since publication on https://google.github.io/comprehensive-rust/ is decoupled from merging the translation into the repository, I suggest we get the suggestions committed and then do a few followup PRs to improve the language.

@KookaS, you should be able to apply the suggestions from Vadim easily. Please look them over when you have some time. When you feel the PR is in a good state, let us know so we can merge it.

File: po/fr.po (hunk 1 above)
Comment (KookaS) [reply po/fr.po:17]: I replaced everywhere with `Comprehensive Rust(le guide complet de Rust)` :)

**Comment by KookaS** [comment]
> Hi all, thanks for the translation and thanks to @vcaen and @sboukortt for jumping in and reviewing this!
> 
> Since publication on https://google.github.io/comprehensive-rust/ is decoupled from merging the translation into the repository, I suggest we get the suggestions committed and then do a few followup PRs to improve the language.
> 
> @KookaS, you should be able to apply the suggestions from Vadim easily. Please look them over when you have some time. When you feel the PR is in a good state, let us know so we can merge it.

I am currently at a bit less than a third of the translation, I'll work on it regular basis. Of course, I'll let you know when it's 100% ready.

**Comment by mgeisler** [comment]
> I am currently at a bit less than a third of the translation, I'll work on it regular basis. Of course, I'll let you know when it's 100% ready.

Thank a lot! It's a lot of work to translate nearly 200 pages :smile: 

I would like us to merge the first third already. Having the `fr.po` file in the repository makes it much easier for other French speakers to come by and translate things.

**Comment by KookaS** [comment]
> > I am currently at a bit less than a third of the translation, I'll work on it regular basis. Of course, I'll let you know when it's 100% ready.
> 
> Thank a lot! It's a lot of work to translate nearly 200 pages smile
> 
> I would like us to merge the first third already. Having the `fr.po` file in the repository makes it much easier for other French speakers to come by and translate things.

Sure, it sounds good to me, I'll keep on working on the translation after the merge.
@vcaen you still need to sign the agreement since I merged your suggestions.

**Comment by mgeisler** [comment]
Hi @KookaS, @vcaen is out on vacation for a few days — but feel free to push more commits to your branch when you have time. Then we'll review and merge it all when he gets back!

**Comment by vcaen** [comment]
I don't understand why the CLA check fails on my email since I'm registered as a Googler with this account. I've filled a bug  internally and will follow up. 

@mgeisler I think you can override the CLA check as the admin of the repo.

**Comment by mgeisler** [comment]
> @KookaS Can you please try running `git filter-branch --msg-filter 'sed "s/vadimcaen@gmail.com/caen@google.com/g"'` in your git repo? That should change the address and fix the CLA issue.

Not needed, the CLA check is fine now that I asked it to rescan the commits!

@KookaS, if you mark the PR as "ready for review", then we can merge it :tada:

**Comment by KookaS** [comment]
After merging, for further translations we open a new branch or we use a fork again?

**Review by mgeisler — APPROVED** [review]
Thanks so much! :rocket:

**Comment by mgeisler** [comment]
> After merging, for further translations we open a new branch or we use a fork again?

You should keep using your fork. Now that this repository has the file as `po/fr.po`, you need to [sync your fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork). You then pull down the `main` branch to your computer, do further updates, push them to your fork and open a new PR. As many times as needed :smile:
